src/map.cpp

```diff
@@ -157,11 +156,11 @@ void Map::removeTile(uint16_t x, uint16_t y, uint8_t z)
 		return;
 	}
 
-	Tile* tile = floor->tiles[x & FLOOR_MASK][y & FLOOR_MASK];
+	std::shared_ptr<Tile> tile = floor->tiles[x & FLOOR_MASK][y & FLOOR_MASK];
 	if (tile) {
 		if (const CreatureVector* creatures = tile->getCreatures()) {
 			for (int32_t i = creatures->size(); --i >= 0;) {
-				if (Player* player = (*creatures)[i]->getPlayer()) {
+				if (auto player = (*creatures)[i]->getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
				if (const auto player = (*creatures)[i]->getPlayer()) {
```